hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,256 @@
+
+    Attributes:
+        callback (``callable`` [`str`, `list`], `dict` ]): The python function
+            that will be called to perform custom shape moves on arbitrary shape
+            parameters. The function must take the particle type and a list of
+            parameters as arguments and return a dictionary with the shape
+            definition whose keys **must** match the shape definition of the
+            integrator:
+
+                ``callable[[str, list], dict]``
+
+            Note that there is no type validation of the callback.
+
+        params (`TypeParameter` [``particle type``, `list`]): List of tunable
+            parameters to be updated.
+
+        move_probability (`float`): Average fraction of shape parameters to
+            change each timestep.
+    """
+
+    _suported_shapes = {
+        'ConvexPolyhedron', 'ConvexSpheropolyhedron', 'Ellipsoid'
+    }
+
+    def __init__(self, move_probability=1):
+
+        super().__init__(move_probability)
+        param_dict = ParameterDict(callback=object)
+        self._param_dict.update(param_dict)
+
+        typeparam_shapeparams = TypeParameter('params',
+                                              type_kind='particle_types',
+                                              param_dict=TypeParameterDict(
+                                                  [float], len_keys=1))
+        self._add_typeparam(typeparam_shapeparams)
+
+    @log(category='object', requires_run=True)
+    def type_params(self):
+        """dict: Tunable parameters as dictionary. The particle types are the
+                 keys of the dictionary.
+
+        Example:
+            >>> python_shape_move.type_params()
+            {'A': [0.21, 0.33], 'B': [0.561, 0.331, 0.123]}
+        """
+        return self._cpp_obj.getTypeParams()
+
+
+class VertexShapeMove(ShapeMove):
+    """Apply shape moves where particle vertices are translated.
+
+    Args:
+        move_probability (float): Average fraction of vertices to change during
+            each shape move.
+
+    Note:
+        Vertices are rescaled during each shape move to ensure that the shape
+        maintains a constant volume. To preserve detail balance, the maximum
+        step size is rescaled by volume**(1/3) every time a move is accepted.
+
+    Note:
+        The shape definition used corresponds to the convex hull of the
+        vertices.
+
+    Example::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456)
+        cube_verts = [(1, 1, 1), (1, 1, -1), (1, -1, 1), (-1, 1, 1),
+                      (1, -1, -1), (-1, 1, -1), (-1, -1, 1), (-1, -1, -1)])
+        mc.shape["A"] = dict(vertices=numpy.asarray(cube_verts) / 2)
+        vertex_move = shape_move.VertexShapeMove()
+        vertex_move.volume["A"] = 1
+
+    Attributes:
+        move_probability (float): Average fraction of vertices to change during
+            each shape move.
+
+        volume (float): Volume of the particles to hold constant.
```

Comment:
What happens when this doesn't line up with the initial volumes of the shapes?

Reply:
The volume is always rescaled upon construction and within the volume setter. Following https://github.com/glotzerlab/hoomd-blue/pull/989#discussion_r846659044, I am not sure if this belong here or to the shape updater class. I've left it here because that is how it was originally implemented. What do you think?

Reply:
I think that we should mention in the attributes section that we always rescale volume of the particles.